### PR TITLE
Fix broken includes when compiling with `builtin_miniupnpc=false`

### LINKED ISSUE #99196: Cannot build with `builtin_miniupnpc=no`
### Tested versions

Latest master

### System information

Godot v4.4.dev (-) - Ubuntu 24.04.1 LTS 24.04 on X11 - X11 display driver, Multi-window, 2 monitors - Vulkan (Mobile) - dedicated AMD Radeon RX 6800 XT (RADV NAVI21) - AMD Ryzen 9 5900X 12-Core Processor (24 threads)

### Issue description

PR #99171 breaks when `builtin_miniupnpc=yes` on Ubuntu 24.04 LTS

This is because:

```bash
pkg-config miniupnpc --cflags --libs
-lminiupnpc
```

Returns those flags, but Godot does:

```cpp
#include <miniupnpc.h>
#include <upnpcommands.h>
// etc
```

However the official package expects the following:

```cpp
#include <miniupnpc/miniupnpc.h>
#include <miniupnpc/upnpcommands.h>
```

Before the PR, it compiled fine because Godot explicitly added:

```py
env.Prepend(CPPPATH=["/usr/include/miniupnpc"])
```

### Steps to reproduce

Compile with:

```bash
pyston-scons platform=linuxbsd builtin_embree=yes builtin_enet=no builtin_freetype=yes builtin_graphite=yes builtin_harfbuzz=yes builtin_libogg=no builtin_libpng=yes builtin_libtheora=no builtin_libvorbis=no builtin_libwebp=no builtin_mbedtls=no builtin_miniupnpc=no builtin_pcre2=no builtin_zlib=yes builtin_zstd=no linker=mold optimize=none debug_symbols=True tests=False dev_mode=True dev_build=True use_llvm=yes use_lld=yes  opengl3=no openxr=no -j 24
```

### Minimal reproduction project (MRP)

N / A

### PR DESCRIPTION
**Pull request by ArchercatNEO**
Fixes https://github.com/godotengine/godot/issues/99196

### CODE REVIEW

File: modules/upnp/SCsub [hunk 1]
```diff
@@ -35,6 +35,7 @@ if env["builtin_miniupnpc"]:
         env_upnp.Append(CPPDEFINES=["MINIUPNPC_SET_SOCKET_TIMEOUT"])
 
     env_thirdparty = env_upnp.Clone()
+    env_thirdparty.Prepend(CPPPATH=[thirdparty_dir + "include/miniupnpc"])
```
Comment (akien-mga) [inline modules/upnp/SCsub:38]: Shouldn't it be this so it resolves the `miniupnpc/` part of the includes?
```suggestion
    env_thirdparty.Prepend(CPPPATH=[thirdparty_dir + "include"])
```

**Review by radiantgurl — APPROVED** [review]
Compiled successfully with `builtin_miniupnpc=false` on Arch Linux 6.11.8-arch1-2

**Review by akien-mga — APPROVED** [review]
Looks good!

**Review by Repiteo — CHANGES_REQUESTED** [review]
Will need a rebase, as well as a minor `README.md` tweak; otherwise, nice work!
```diff
 thirdparty/README.md | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/thirdparty/README.md b/thirdparty/README.md
index 7a95e3c724..760ff0a1e6 100644
--- a/thirdparty/README.md
+++ b/thirdparty/README.md
@@ -628,7 +628,8 @@ to solve some MSVC warnings. See the patches in the `patches` directory.
 
 Files extracted from upstream source:
 
-- Copy `miniupnpc/src` and `miniupnpc/include` to `thirdparty/miniupnpc`
+- `miniupnpc/src/` as `src/`
+- `miniupnpc/include/` as `include/miniupnpc/`
 - Remove the following test or sample files:
   `listdevices.c,minihttptestserver.c,miniupnpcmodule.c,upnpc.c,upnperrors.*,test*`
 - `LICENSE`
```

**Review by akien-mga — APPROVED** [review]

**Review by Repiteo — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks! And congrats for your first merged Godot contribution :tada:

**Pull request merged**